### PR TITLE
Update `GetSchema` filtering to exclude shards where `IsMasterServing` but no `MasterAlias`

### PR DESCRIPTION
**Pull request by ajm188**
Signed-off-by: Andrew Mason <amason@slack-corp.com>

## Description

This effectively accounts for corner cases where there is an empty shard which happens to also be marked as serving.

## Related Issue(s)

- 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [x]  VTAdmin

### CODE REVIEW

File: go/vt/vtadmin/cluster/cluster.go [hunk 1]
```diff
@@ -633,7 +633,7 @@ func (c *Cluster) getTabletsToQueryForSchemas(ctx context.Context, keyspace stri
 		tabletsToQuery := make([]*vtadminpb.Tablet, 0, len(shards))
 
 		for _, shard := range shards {
-			if !shard.Shard.IsMasterServing {
+			if !(shard.Shard.IsMasterServing && shard.Shard.MasterAlias != nil) {
```
Comment (rafael) [inline go/vt/vtadmin/cluster/cluster.go:636]: Could you add comment for why we need to do this check? Our future selves will be thankful for that :D

**Review by rafael — APPROVED** [review]
LGTM

**Comment by rafael** [comment]
Looks good, let's merge when the tests pass.
